command/agentproxyshared/cache/lease_cache.go

```diff
@@ -586,6 +734,14 @@ func computeIndexID(req *SendRequest) (string, error) {
 	return hex.EncodeToString(cryptoutil.Blake2b256Hash(string(b.Bytes()))), nil
 }
 
+// computeStaticSecretCacheIndex results in a value that uniquely identifies a static
+// secret's cached ID. Notably, we intentionally ignore headers (for example,
+// the X-Vault-Token header) to remain agnostic to which token is being
+// used in the request. We care only about the path.
+func computeStaticSecretCacheIndex(req *SendRequest) string {
+	return hex.EncodeToString(cryptoutil.Blake2b256Hash(req.Request.URL.Path))
```

Comment:
Shouldn't we care about the namespace header though?

Reply:
Hmm, that's a good point. We'll have to ensure that the namespace header and the path get added together, e.g. a request to `ns1/secret` should be the same as a request to `secret` with `ns1` as the namespace header. This is actually a shortcoming of the existing approach for dynamic secrets